examples/widgets/check_buttons.py

```diff
@@ -25,21 +25,23 @@
 l2, = ax.plot(t, s2, lw=2, color='g', label='6 Hz')
 fig.subplots_adjust(left=0.2)
 
-lines = [l0, l1, l2]
+lines = {l.get_label(): l for l in [l0, l1, l2]}
 
 # Make checkbuttons with all plotted lines with correct visibility
 rax = fig.add_axes([0.05, 0.4, 0.1, 0.15])
-labels = [str(line.get_label()) for line in lines]
-visibility = [line.get_visible() for line in lines]
-check = CheckButtons(rax, labels, visibility)
+check = CheckButtons(
+    ax=rax,
+    labels=list(lines),
+    actives=[l.get_visible() for l in lines.values()]
```

Comment:
I don't think making `lines` a dict makes this necessarily better. It adds the cognitive load to rember what keys and values are (which is not not clear from the name `lines`.

I would leave `lines = [l0, l1, l2]`, which is very simple, and instead inline the comprehensions:

```suggestion
    labels=[l.get_label() for l in lines]
    actives=[l.get_visible() for l in lines]
```
This is also nicely symmetric in code structure.